src/cmd-buildextend-installer

```diff
@@ -235,12 +255,46 @@ def generate_iso():
     if arch == "x86_64":
         run_verbose(['/usr/bin/isohybrid', tmpisofile])
 
+    # Find the byte offset of the initrd padding within the ISO image
+    # and write it into a custom header at the end of the ISO 9660 System
+    # Area.  This tells coreos-iso-embed-ignition where to write a
+    # cpio.gz with a custom Ignition config.  Skip on s390x because
+    # it uses an embedded El Torito image with its own copy of the initramfs.
+    if is_live and arch != "s390x":
+        isoinfo = run_verbose(['isoinfo', '-lR', '-i', tmpisofile],
+                stdout=subprocess.PIPE, text=True)
+        matches = list(re.finditer(r'\[\s*([0-9]+) 00\]\s+initramfs\.img\s*$',
+                isoinfo.stdout, re.MULTILINE))
+        if len(matches) != 1:
+            raise Exception('Found {} copies of initramfs.img'.format(len(matches)))
+        offset = int(matches[0].group(1)) * 2048  # bytes per sector
+        offset += os.stat(os.path.join(tmpisoimages, "initramfs.img")).st_size - initrd_ignition_padding
+        with open(tmpisofile, 'r+b') as isofh:
+            # Verify that the calculated byte range is empty
+            isofh.seek(offset)
+            if isofh.read(initrd_ignition_padding) != bytes(initrd_ignition_padding):
+                raise Exception(f'ISO image {initrd_ignition_padding} bytes at {offset} are not zero')
+            # Write header
+            fmt = '<8s2Q'
+            isofh.seek(32768 - struct.calcsize(fmt))
+            isofh.write(struct.pack(fmt, b'coreiso+', offset, initrd_ignition_padding))
+            print(f'Embedded {initrd_ignition_padding} bytes Ignition config space at {offset}')
+
     kernel_name = f'{base_name}-{args.build}-{image_type}-kernel'
     initramfs_name = f'{base_name}-{args.build}-{image_type}-initramfs.img'
     kernel_file = os.path.join(builddir, kernel_name)
     initramfs_file = os.path.join(builddir, initramfs_name)
     shutil.copyfile(os.path.join(tmpisoimages, "vmlinuz"), kernel_file)
-    shutil.copyfile(os.path.join(tmpisoimages, "initramfs.img"), initramfs_file)
+    with open(os.path.join(tmpisoimages, "initramfs.img"), 'rb') as fsrc:
+        with open(initramfs_file, 'wb') as fdst:
+            shutil.copyfileobj(fsrc, fdst)
+            if is_live:
+                # Verify ISO initrd padding and truncate it away
```

Comment:
I don't expect anyone will want to unpack the ISO and checksum the initramfs image inside it.  The idea is to avoid 256 KB of extra download/storage for use cases that don't need it.

I just tried, and it appears that it's still possible to append to a padded initrd image.  That surprises me, though.